source/docs/software/commandbased/structuring-command-based-project.rst

```diff
@@ -1,239 +1,213 @@
+      :language: java
+      :lines: 63-74
+      :linenos:
+      :lineno-start: 63
+
+  .. tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/templates/commandbased/cpp/Robot.cpp
+      :language: c++
+      :lines: 34-44
+      :linenos:
+      :lineno-start: 34
+
+The ``autonomousInit()`` method schedules an autonomous command returned by the ``RobotContainer`` instance. The logic for selecting which autonomous command to run can be handled inside of ``RobotContainer``.
+
+.. tabs::
+
+  .. tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/templates/commandbased/Robot.java
+      :language: java
+      :lines: 83-92
+      :linenos:
+      :lineno-start: 83
+
+  .. tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/templates/commandbased/cpp/Robot.cpp
+      :language: c++
+      :lines: 48-57
+      :linenos:
+      :lineno-start: 48
+
+The ``teleopInit()`` method cancels any still-running autonomous commands. This is generally good practice.
+
+Advanced users are free to add additional code to the various init and periodic methods as they see fit; however, it should be noted that including large amounts of imperative robot code in ``Robot.java`` is contrary to the declarative design philosophy of the command-based paradigm, and can result in confusingly-structured/disorganized code.
+
+RobotContainer
 --------------
 
-The ``Constants.java`` class is where globally-accessible robot
-constants (such as speeds, unit conversion factors, PID gains, and
-sensor/motor ports) can be stored. It is recommended that users separate
-these constants into individual inner classes corresponding to subsystems
-or robot modes, to keep variable names shorter. All constants declared
-in ``Constants.java`` should be declared as ``public static final`` so
-that they are globally accessible and cannot be changed.
+This class (`Java <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/templates/commandbased/RobotContainer.java>`__, `C++ (Header) <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/templates/commandbased/include/RobotContainer.h>`__, `C++ (Source) <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/templates/commandbased/cpp/RobotContainer.cpp>`__) is where most of the setup for your command-based robot will take place. In this class, you will define your robot’s subsystems and commands, bind those commands to triggering events (such as buttons), and specify which command you will run in your autonomous routine. There are a few aspects of this class new users may want explanations for:
+
+.. tabs::
+
+  .. tab:: Java
```

Comment:
Should these be tab groups (I think that is the name) so all the tabs change for the whole page?